src/main/java/org/broadinstitute/hellbender/tools/walkers/annotator/VariantAnnotatorEngine.java

```diff
@@ -0,0 +1,263 @@
+package org.broadinstitute.hellbender.tools.walkers.annotator;
+
+import com.google.common.collect.Sets;
+import htsjdk.variant.variantcontext.*;
+import htsjdk.variant.vcf.*;
+import org.broadinstitute.hellbender.engine.FeatureContext;
+import org.broadinstitute.hellbender.engine.FeatureInput;
+import org.broadinstitute.hellbender.engine.ReferenceContext;
+import org.broadinstitute.hellbender.exceptions.UserException.BadArgumentValue;
+import org.broadinstitute.hellbender.tools.walkers.annotator.interfaces.AnnotationGroup;
+import org.broadinstitute.hellbender.tools.walkers.annotator.interfaces.GenotypeAnnotation;
+import org.broadinstitute.hellbender.tools.walkers.annotator.interfaces.InfoFieldAnnotation;
+import org.broadinstitute.hellbender.tools.walkers.annotator.interfaces.VariantAnnotation;
+import org.broadinstitute.hellbender.utils.Utils;
+import org.broadinstitute.hellbender.utils.genotyper.PerReadAlleleLikelihoodMap;
+
+import java.util.*;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static org.broadinstitute.hellbender.tools.walkers.annotator.interfaces.AnnotationGroup.ActiveRegionBased;
+import static org.broadinstitute.hellbender.tools.walkers.annotator.interfaces.AnnotationGroup.Standard;
+
+/**
+ * The class responsible for computing annotations for variants.
+ */
+public final class VariantAnnotatorEngine {
+    private final List<InfoFieldAnnotation> infoAnnotations;
+    private final List<GenotypeAnnotation> genotypeAnnotations;
+
+    private final VariantOverlapAnnotator variantOverlapAnnotator;
+
+    private VariantAnnotatorEngine(final AnnotationManager annots, final FeatureInput<VariantContext> dbSNPBinding){
```

Comment:
Or is this actually intended to be SPECIFICALLY a DBSNP file, and not some other set of variants?